pkg/paths/paths.go

```diff
@@ -65,13 +65,18 @@ func (p Path) HasPrefix(prefix Path) bool {
 func (p Path) Encrypt(key []byte) (encrypted Path, err error) {
 	encrypted = make([]string, len(p))
 	for i, seg := range p {
-		encrypted[i], err = encrypt(seg, key)
-		if err != nil {
-			return nil, err
-		}
-		key, err = deriveSecret(key, seg)
-		if err != nil {
-			return nil, err
+		// TODO(moby) this is a hacky way to deal with buckets
```

Comment:
_you're a hacky way to deal with buckets_

Reply:
https://tenor.com/pbrU.gif